driver-core/src/main/com/mongodb/connection/netty/NettyStream.java

```diff
@@ -336,4 +341,47 @@ public T get() throws IOException {
             }
         }
     }
+
+
+    private void scheduleReadTimeout() {
+        adjustTimeout(false);
+    }
+
+    private void disableReadTimeout() {
+        adjustTimeout(true);
+    }
+
+    private void adjustTimeout(final boolean disable) {
+            ChannelHandler timeoutHandler = channel.pipeline().get(READ_HANDLER_NAME);
+            if (timeoutHandler != null && timeoutHandler instanceof ReadTimeoutHandler) {
```

Comment:
timeoutHandler can be null - as we may or may not add it, now.  If we dont then continue as normal.